src/main/java/io/spokestack/spokestack/Spokestack.java

```diff
@@ -0,0 +1,659 @@
+package io.spokestack.spokestack;
+
+import android.content.Context;
+import androidx.lifecycle.Lifecycle;
+import io.spokestack.spokestack.nlu.NLUResult;
+import io.spokestack.spokestack.nlu.NLUService;
+import io.spokestack.spokestack.nlu.tensorflow.TensorflowNLU;
+import io.spokestack.spokestack.tts.SynthesisRequest;
+import io.spokestack.spokestack.tts.TTSManager;
+import io.spokestack.spokestack.util.AsyncResult;
+import io.spokestack.spokestack.util.EventTracer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.spokestack.spokestack.SpeechPipeline.DEFAULT_BUFFER_WIDTH;
+import static io.spokestack.spokestack.SpeechPipeline.DEFAULT_FRAME_WIDTH;
+import static io.spokestack.spokestack.SpeechPipeline.DEFAULT_SAMPLE_RATE;
+
+/**
+ * This class combines all Spokestack subsystems into a single component to
+ * provide a unified interface to the library's ASR, NLU, and TTS features. Like
+ * the individual subsystems, it is configurable using a fluent builder pattern,
+ * but it provides a default configuration; only a few parameters are required
+ * from the calling application, and those only for specific features noted in
+ * the documentation for the builder's methods.
+ *
+ * <p>
+ * Client applications may wish to establish event listeners for purposes such
```

Comment:
What would including logging here look like? The doc here is referring to `onTrace` events for the NLU and the similar speech event for the speech pipeline, so that those types of events can be forwarded to whatever logging system the app is using. I'm not interested in the complexity of adding Android system logging hooks here, if that's what you mean.

Reply:
Again, perhaps better docs to resolve the difference between the Spokestack system logging (eg `onTrace`) and Android system logging.